### PR TITLE
DOS-Tests for digital outcomes, user research studios and participants service submission

### PR DESCRIPTION
**Pull request by andrewchong**
First attempt at tests for submitting services. Further scenarios to come.
Refactored and made changes to current G-Cloud SSP tests to account for flow changes and DOS service submission, allowing reuse of code between the two frameworks.

### CODE REVIEW

File: features/DOS/ssp_digital_outcomes_service.feature [hunk 1]
```diff
@@ -0,0 +1,78 @@
+@not-production @functional-test @ssp
```
Comment (TheDoubleK) [inline features/DOS/ssp_digital_outcomes_service.feature]: Should this perhaps be `@ssp-dos`, seeing as we also have `@ssp-g-cloud` now?

File: features/step_definitions/ssp_steps.rb [hunk 2]
```diff
@@ -22,15 +22,23 @@
 end
 
 Given /^I am at the g7 services page$/ do
-  visit "#{dm_frontend_domain}/suppliers/frameworks/g-cloud-7/services"
+  #visit "#{dm_frontend_domain}/suppliers/frameworks/g-cloud-7/services"
```
Comment (TheDoubleK) [inline features/step_definitions/ssp_steps.rb]: These commented-out lines can probably be deleted if the generalised version is working OK?

File: features/step_definitions/ssp_steps.rb [hunk 3]
```diff
@@ -75,11 +106,19 @@
 end
 
 Then /^I should be on the g7 supplier dashboard page$/ do
```
Comment (TheDoubleK) [inline features/step_definitions/ssp_steps.rb]: Should this be changed to something like `I should be on the framework dashboard page`, as it is no longer G7-specific.

File: features/step_definitions/ssp_steps.rb [hunk 4]
```diff
@@ -75,11 +106,19 @@
...
-  URI.parse(current_url).path.should == "/suppliers/frameworks/g-cloud-7"
+  #URI.parse(current_url).path.should == "/suppliers/frameworks/g-cloud-7"
+  URI.parse(current_url).path.should == "/suppliers/frameworks/#{store.framework_name}"
 end
 
 Then /^I should be on the g7 services page$/ do
```
Comment (TheDoubleK) [inline features/step_definitions/ssp_steps.rb]: Also here - something like `I should be on the framework services page`

File: features/step_definitions/ssp_steps.rb [hunk 5]
```diff
@@ -22,15 +22,23 @@
 end
 
 Given /^I am at the g7 services page$/ do
```
Comment (TheDoubleK) [inline features/step_definitions/ssp_steps.rb]: This is another name that could change if the method isn't only for G7 now.

**Comment by TheDoubleK** [comment]
Few minor comments but otherwise looks good.

**Comment by andrewchong** [comment]
@TheDoubleK: I have made changes as per your comments. Can you please have another look when you are able to.

**Comment by TheDoubleK** [comment]
Looks good.  The only other thing I notice is that `g-cloud_supplier_creation.feature` isn't really G-Cloud specific - it's the same account for whatever framework, so could probably be re-named and moved out of the "G-Cloud" folder.

But it probably doesn't matter much either way, and if you think that that journey belongs with the rest of the G-Cloud tests then that's fine.

:+1: either before or after the move of this feature file, just something to think about.

**Pull request merged**